src/env/node/git/localGitProvider.ts

```diff
@@ -1839,6 +1843,23 @@ export class LocalGitProvider implements GitProvider, Disposable {
 				});
 			}
 
+			const status: GitStatus | undefined = await this.getStatusForRepo(repoPath);
+			const workingTreeStatus = status?.computeWorkingTreeStatus() || undefined;
+			if (workingTreeStatus?.staged || workingTreeStatus?.unstaged) {
+				rows.unshift({
+					sha: 'work-dir-changes',
+					parents: headRefSha === undefined ? [] : [headRefSha],
+					author: '',
+					email: '',
+					date: new Date().getTime(),
+					message: '',
+					type: GitGraphRowType.Working,
+					heads: [],
+					remotes: [],
+					tags: [],
+				});
+			}
+
```

Comment:
This needs to be moved into the outer function -- since we only want this called once when we get the original set of rows, not on each page.

This is also a "refresh"/performance challenge -- since it is much more likely that the WIP will change than all the rows. So I think we might actually have to move this out the provider altogether, and instead add it into `graphWebview` so that it could not only be included in the original rows payload, but we could have a specific single row update when it changes

Reply:
@eamodio What if I just checked (in the `if` clause on 1848) if no cursor has been set?

i.e...
```
if (!cursor && (workingTreeStatus?.staged || workingTreeStatus?.unstaged)) {
...
```

That would at least fix the "only on the first set" issue, but the performance issue you mentioned would still be there I suppose.

In that case we may need separate state update logic that adds/removes/updates the date on the WIP row.

Reply:
Yeah -- we'd have to plumb through a new way to splice in just that row (at least from the GitLens -> webview side -- we could just pass the full set of rows to the component again

Reply:
I think I have an idea. What if I just add in the WIP node on the component side? We already have a prop for work dir stats that can be passed in from GitLens. We could just listen for workdir changes and update that prop and pass it through from this side, and on the component side, staple a WIP node to the top whenever the workdirchanges prop is non-empty, and update that node whenever the prop changes. WDYT?

Reply:
Sounds great! We'd still need to move the workdir calc out of the provider and into the `grapWebview` or add a new wrapper method on the provider to give out the data (and events) that we need -- though I think that might be overkill